src/org/zaproxy/zap/extension/ascanrulesAlpha/IntegerOverflow.java

```diff
@@ -0,0 +1,234 @@
+	}
+	
+	private String getError(char c) {
+		return Constant.messages.getString(MESSAGE_PREFIX + "error"+c);
+	}
+	
+	@Override
+	public void init() {
+
+	}
+
+	/*
+	 * This method is called by the active scanner for each GET and POST parameter for every page 
+	 * @see org.parosproxy.paros.core.scanner.AbstractAppParamPlugin#scan(org.parosproxy.paros.network.HttpMessage, java.lang.String, java.lang.String)
+	 */
+	@Override
+	public void scan(HttpMessage msg, String param, String value) {
+	
+		if (checkStop() == true) {
+			return;
+		}
+			
+			String returnAttack = randomIntegerString(4);// The number of full length ints to send.		
+			if (attackVector(msg, param, '1', returnAttack) == true) {
+				return;
+			}
+			returnAttack = singleString(4,'0'); // The number of full length ints to send.
+			if (attackVector(msg, param, '2', returnAttack) == true) {
+				return;
+			}
+			returnAttack = singleString(4,'1');// The number of full length ints to send.
+			if (attackVector(msg, param, '3', returnAttack) == true) {
+				return;
+			}
+			returnAttack = singleString(4,'9');// The number of full length ints to send.
+			if (attackVector(msg, param, '4', returnAttack) == true) {
+				return;
+			}
+				return;	
+
+	}
+
+	@Override
+	public int getRisk() {
+		return Alert.RISK_MEDIUM;
+	}
+
+	@Override
+	public int getCweId() {
+		// The CWE id
+		return 190;
+	}
+
+	@Override
+	public int getWascId() {
+		// The WASC ID
+		return 3;
+	}
+	
+	private String randomIntegerString(int length)
+	{
+		
+		int numbercounter = 0;
+		int character = 0;
+		long charactercounter = 0;
+		int maxcharacter = 11;
+		StringBuilder sb1 = new StringBuilder(maxcharacter*length);
+		while (numbercounter < length)
+		{
+			charactercounter = 0;
+			while (charactercounter < maxcharacter)
+			{	
+				 character = 48 + (int) (Math.random()*10); 
+	
+			        while ( character > 57 && character < 48)
+			        {
+			        	character = 48 + (int) (Math.random()*10); 
+			        }
+	
+					charactercounter++;
+					sb1.append((char)character);
+			}
+			numbercounter++;
+		}
+		return sb1.toString();
+	}
+	
+	private String singleString(int length, char c)//Single Character String
+	{
+		
+		int numbercounter = 0;
+		long charactercounter = 0;
+		int maxcharacter = 11;
+		StringBuilder sb1 = new StringBuilder(maxcharacter*length);
+		while (numbercounter < length)
+		{	
+			charactercounter = 0;
+			while (charactercounter < maxcharacter)
+			{	
+					charactercounter++;
+					sb1.append(c);
+			}
+			numbercounter++;
+		}
+		return sb1.toString();
+	}
+	
+	private boolean checkStop()
+	{
+		if (this.isStop()) { // Check if the user stopped things
+			if (log.isDebugEnabled()) {
+				log.debug("Scanner "+this.getName()+" Stopping.");
+			}
+			return true; // Stop!
+		}
+		return false;
+	}
+	private boolean attackVector(HttpMessage msg, String param,char type,String returnAttack)
+	{
+		if (checkStop() == true) {
+			return true;
+		}
+		msg = getNewMsg();

```

Comment:
It could be defined a local variable instead (the `HttpMessage` parameter could be removed).